### PR TITLE
Fix regex in tdt is email method

### PR DESCRIPTION
**Pull request by pratvick**

### CODE REVIEW

**Comment by pratvick** [comment]
@yatinsns Please review this

**Comment by yatinsns** [comment]
@mx4492 Please review this.

**Comment by mx4492** [comment]
Since other teams are using this regex, this change would require building up consensus and will not be backward compatible. @pratvick please create a new method with your changes and leave the old one as it is. If you can't come up with a better name, then `tdt_isEmail2` is fine too; just mention in the header docs that this will use a simpler regex than the existing one.

**Comment by pratvick** [comment]
@mx4492 Please review.

File: TDTChocolate/FoundationAdditions/NSString+TDTAdditions.h [hunk 1]
```diff
@@ -78,6 +78,12 @@
 - (BOOL)tdt_isEmail;
 
 /**
+ This method is using the simpler regex than the regex in the existing method.
```
Comment (mx4492) [inline TDTChocolate/FoundationAdditions/NSString+TDTAdditions.h]:
```
Variant of @pre tdt_isEmail that uses a simpler regex with the aim:
"Never disallow a valid email address, and disallow _most_ invalid ones"
```

**Comment by mx4492** [comment]
GTM otherwise.

**Pull request merged**